x/programs/test/util.go

```diff
@@ -84,7 +84,7 @@ func prependAccountToKey(account codec.Address, key []byte) []byte {
 
 func Into[T any](data []byte) T {
 	result := new(T)
-	if err := borsh.Deserialize(result, data); err != nil {
+	if err := borsh.Deserialize(result, data[1:]); err != nil {
```

Comment:
Wait... what happened here?

Reply:
This is because the data is used from a function call, which contains this 1 or 0 prepended byte for `Ok` or `Err`. But it definitely should not be here

Reply:
I almost want to write an `unwrap` function...

Reply:
That's what I've done, let's see if you are happy with it 😄

Reply:
But how do you know what you're deserializing into? I'm not suggesting that this code is wrong, it's just not clear _why_ it's right

Reply:
I removed the access to `[1:]` and wrote this `Unwrap` function instead, to be used after the function call to the program. Is that what you are talking about ?